libraries/AP_Logger/AP_Logger.cpp

```diff
@@ -504,10 +511,16 @@ void AP_Logger::backend_starting_new_log(const AP_Logger_Backend *backend)
 
 bool AP_Logger::should_log(const uint32_t mask) const
 {
+    bool armed = vehicle_is_armed();
+
+    if (armed) {
+        _last_armed_ms = AP_HAL::millis();
```

Comment:
Why isn't this in `set_vehicle_armed`?  It's really, really, not my favourite methodcall, but it is currently there.  This would allow `_last_armed_ms` to be non-static.

Reply:
Also makes the timer not be dependent on when should_log is called, but instead on the real disarm time.